server/plugin/api.go

```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
 	p.writeJSON(w, resp)
 }
 
+func (p *Plugin) updatePost(post *model.Post, issue *IssueRequestToUpdate, w http.ResponseWriter) {
+	post, appErr := p.API.GetPost(issue.PostID)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s", issue.PostID), StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s : not found", issue.PostID), StatusCode: http.StatusNotFound})
+		return
+	}
+
+	post.Props[AssigneesForProps] = issue.Assignees
+	post.Props[LabelsForProps] = issue.Labels
+	post.Props[DescriptionForProps] = issue.Body
+	post.Props[TitleForProps] = issue.Title
+	if _, appErr = p.API.UpdatePost(post); appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to update post " + issue.PostID, StatusCode: http.StatusInternalServerError})
+	}
+}
+
+func (p *Plugin) validateIssueRequestToUpdate(issue *IssueRequestToUpdate, w http.ResponseWriter) bool {
+	if issue.Title == "" {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue title.", StatusCode: http.StatusBadRequest})
+		return false
+	}
+	if issue.PostID == "" && issue.ChannelID == "" {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide either a postID or a channelID", StatusCode: http.StatusBadRequest})
+		return false
+	}
+
+	return true
+}
+
+func (p *Plugin) updateIssue(c *UserContext, w http.ResponseWriter, r *http.Request) {
+	// get data for the issue from the request body and fill IssueRequestToUpdate object so that we can update the issue
+	issue := &IssueRequestToUpdate{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		c.Log.WithError(err).Warnf("Error decoding the JSON body")
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid JSON object.", StatusCode: http.StatusBadRequest})
```

Comment:
do we need to send the actual error as well to the API response?

Reply:
I don't this soo.. as we are sending the message.